hallucinogenizer/quiz-app-2
# Auto assign a Quiz to Applicants

## How to update database
- In `initialize.js`, make sure to temporarily uncomment the following two lines:
    ```javascript
            await ApplicationRound.sync({ alter: true, force: false })
		await Quiz.sync({ alter: true, force: false })
    ```
- Then run the server once so that database is updated. After running it, re-comment the above lines.
- Advice for future: In the future, let's never run global `sequelize.sync({alter:true})`. Only run `sync()` on the specific database modal that has been changed to avoid unintended changes in other models and data.

## How to verify
- Go to the admin panel
- Go to Applications menu
- On any Application's tile there will appear a new icon (right-most). Click it and a modal will open.
- Pick a Quiz to auto-assign.
- Now solve this application in new incognito tab. Create a new student or use an existing one.
- [x] Log in as student after filling the application. The quiz should have been auto assigned to you.
- [ ] I refactored the Application Receipt sending code in the afterSave hook. Do verify that receipts are still sent.
- [ ] Also verify that an application that does not have any auto-assign quiz, i.e. a round whose `auto_assigned_quiz_id` property is `NULL` still works, i.e. a student can still successfully fill such an application.

## To Do
- "Remove Auto Assignment" button in the modal doesn't do anything yet. Make it set the `ApplicationRound.auto_assigned_quiz_id` property to NULL.
